src/YarpUtilities/src/VectorsCollectionServer.cpp

```diff
@@ -0,0 +1,176 @@
+/**
+ * @file VectorsCollectionServer.cpp
+ * @authors Giulio Romualdi
+ * @copyright 2023 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the BSD-3-Clause license.
+ */
+
+#include <BipedalLocomotion/TextLogging/Logger.h>
+#include <BipedalLocomotion/YarpUtilities/VectorsCollectionServer.h>
+
+#include <yarp/os/BufferedPort.h>
+#include <yarp/os/Port.h>
+
+#include <unordered_set>
+
+using namespace BipedalLocomotion::YarpUtilities;
+
+struct VectorsCollectionServer::Impl
+{
+    yarp::os::BufferedPort<VectorsCollection> port; /**< Buffered port used to communicate with
+                                                            the client. */
+    yarp::os::Port rpcPort; /**< RPC port used to communicate with the client. */
+
+    VectorsCollectionMetadata metadata; /**< Metadata of the vectors collection. */
+
+    bool isMetadataFinalized{false}; /**< True if the metadata has been finalized. */
```

Comment:
I would suggest using an atomic variable here to avoid concurrency issues with clients that get connected before the population of the metadata is finalized